### PR TITLE
New features for langium-sprotty

### PR DESCRIPTION
**Pull request by spoenemann**
- Tracing utility: attach tracing information to generated diagram model elements
- Selection synchronization: when you select an element in the diagram, the corresponding source text is selected; when the cursor is put on a DSL element, the corresponding diagram element is selected and centered.
- Diagnostic markers: Attach markers to diagram model elements for which the source has some diagnostic item; hover the mouse over the markers in the diagram to see the diagnostic messages.

### CODE REVIEW

**Review by montymxb — COMMENTED** [review]
Couple of points.

File: packages/langium-sprotty/src/diagram-server-manager.ts [hunk 1]
```diff
@@ -66,45 +73,60 @@ export class DefaultDiagramServerManager implements DiagramServerManager {
...
+            }
             documents.delete(document);
         }
     }
 
+    /**
+     * Determine whether we should actually update the given document. The default implementation skips
+     * updating if there are lexer errors or parser errors.
+     */
+    protected shouldUpdateDiagram(document: LangiumDocument, _language: LangiumSprottyServices): boolean {
```
[the remark this answers is not in the document]
Comment (spoenemann) [reply packages/langium-sprotty/src/diagram-server-manager.ts:120]: This is intended as a template method that users can override to implement more complex logic to determine whether a diagram should be updated or not. The second parameter gives users the option to access language-specific services if they need to.

**Review by montymxb — COMMENTED** [review]
:octocat: LGTM

**Review by montymxb — APPROVED** [review]
Whoops, defaulted to comment. Here's an actual approval

**Pull request merged**